File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        letters = {}
        result = 0
        odd_flag = False
        for ch in s:
            if ch in letters:
                letters[ch] += 1
            else:
                letters[ch] = 1
        for ch, value in letters.items():
            if value % 2 == 0:
                result += value
            else:
                result += (value // 2) * 2;
                odd_flag = True
    
        if odd_flag:
            result += 1
    
        return result

File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int[] low = new int[26];
        int[] upp = new int[26];
        char ch;
        for (int i = 0; i < s.length(); i++) {
            ch = s.charAt(i);
            if (ch >= 'a' && ch <= 'z') {
                low[ch - 'a']++;
            } else {
                upp[ch - 'A']++;
            }
        }
        boolean odd = false;
        int cnt = 0;
        for (int i = 0; i < 26; i++) {
            if (low[i] >= 2) {
                cnt += low[i] / 2;
            }
            if (low[i] % 2 == 1) {
                odd = true;
            }
            if (upp[i] >= 2) {
                cnt += upp[i] / 2;
            }
            if (upp[i] % 2 == 1) {
                odd = true;
            }
        }
        
        return 2 * cnt + (odd ? 1 : 0);
    }
}